From 8dc0a6ca504b823c79c406194da49389caf08828 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 18 Mar 2014 09:18:33 -0700
Subject: [PATCH 477/827] media: xilinx: vip: Add xvip_get_frame_size()

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

xvip_get_frame_size() reads the active size and returns the width and
height.

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
Acked-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit b6fb853081924747839685fdb318ca521efbf09a)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vip.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-vip.h b/drivers/media/platform/xilinx/xilinx-vip.h
index 6f0f945..662c3f5 100644
--- a/drivers/media/platform/xilinx/xilinx-vip.h
+++ b/drivers/media/platform/xilinx/xilinx-vip.h
@@ -168,6 +168,18 @@ static inline void xvip_set_frame_size(struct xvip_device *xvip,
 		   (format->width << XVIP_ACTIVE_HSIZE_SHIFT));
 }
 
+static inline void xvip_get_frame_size(struct xvip_device *xvip,
+				       struct v4l2_mbus_framefmt *format)
+{
+	u32 reg;
+
+	reg = xvip_read(xvip, XVIP_ACTIVE_SIZE);
+	format->width = (reg & XVIP_ACTIVE_HSIZE_MASK) >>
+			XVIP_ACTIVE_HSIZE_SHIFT;
+	format->height = (reg & XVIP_ACTIVE_VSIZE_MASK) >>
+			 XVIP_ACTIVE_VSIZE_SHIFT;
+}
+
 static inline void xvip_enable_reg_update(struct xvip_device *xvip)
 {
 	xvip_set(xvip, XVIP_CTRL_CONTROL, XVIP_CTRL_CONTROL_REG_UPDATE);
-- 
2.9.3

